File: backend/FinanceApp.Domain/Entities/Payable.cs
using FinanceApp.Domain.Enums;
using FinanceApp.Domain.Interfaces;

namespace FinanceApp.Domain.Entities;

public class Payable : BaseEntity, IConcurrencyVersioned
{
    /// <summary>乐观并发版本号（由 AppDbContext 自动维护）</summary>
    public long Version { get; set; }

    public long? SupplierId { get; set; }
    public long? CustomerId { get; set; }
    public long? PersonId { get; set; }
    public long? ProjectId { get; set; }
    public long? PayableTypeId { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal PaidAmount { get; set; }
    public decimal RemainingAmount { get; set; }
    public DateTime? DueDate { get; set; }
    public PayableStatus Status { get; set; } = PayableStatus.Pending;
    public string? Description { get; set; }
    public DateTime? SettledAt { get; set; }

    // Navigation properties
    public Supplier? Supplier { get; set; }
    public Customer? Customer { get; set; }
    public Person? Person { get; set; }
    public Project? Project { get; set; }
    public PayableType? PayableType { get; set; }
    public ICollection<PayableDetail> Details { get; set; } = new List<PayableDetail>();
    public ICollection<TagBinding> TagBindings { get; set; } = new List<TagBinding>();
}

File: frontend/src/features/finance/types/payable.ts
import type { TagItem } from '@/features/master-data/projects/types/project'

export interface Payable {
  id: number
  supplierId?: number
  supplierName?: string
  customerId?: number
  customerName?: string
  personId?: number
  personName?: string
  projectId?: number
  projectName?: string
  payableTypeId?: number
  payableTypeName?: string
  totalAmount: number
  paidAmount: number
  remainingAmount: number
  dueDate?: string
  status: 'pending' | 'partial' | 'settled'
  description?: string
  tags: TagItem[]
  createdAt: string
  updatedAt: string
  settledAt?: string
  details: PayableDetail[]
}

export interface PayableDetail {
  id: number
  payableId: number
  transactionId: number
  paymentDate: string
  amount: number
  paymentMethod?: string
  description?: string
  createdAt: string
}

export interface PayableSummary {
  totalPayable: number
  totalPaid: number
  totalRemaining: number
  pendingCount: number
  partialCount: number
  settledCount: number
  overdueCount: number
}

export interface PayableTrend {
  months: string[]
  amounts: number[]
}

export interface PayableAging {
  categories: string[]
  amounts: number[]
}

export interface CreatePayableRequest {
  supplierId?: number
  customerId?: number
  personId?: number
  projectId?: number
  payableTypeId?: number
  totalAmount: number
  dueDate?: string
  description?: string
}

export interface UpdatePayableRequest {
  supplierId?: number
  customerId?: number
  personId?: number
  projectId?: number
  payableTypeId?: number
  totalAmount: number
  dueDate?: string
  description?: string
}

export interface PayPaymentRequest {
  paymentDate: string
  amount: number
  paymentMethod?: string
  description?: string
  transactionId: number
}

export interface PayableStatistics {
  totalCount: number
  pendingCount: number
  partialCount: number
  settledCount: number
  totalAmount: number
  paidAmount: number
  remainingAmount: number
  overdueAmount: number
}

export interface PayableType {
  id: number
  name: string
  code?: string
  description?: string
  isActive: boolean
  sortOrder: number
}

export interface CreatePayableTypeRequest {
  name: string
  code?: string
  description?: string
  isActive: boolean
  sortOrder: number
}

export interface UpdatePayableTypeRequest {
  name: string
  code?: string
  description?: string
  isActive: boolean
  sortOrder: number
}
